rocky/views/ooi_list.py

```diff
@@ -24,9 +36,78 @@ def get_context_data(self, **kwargs):
 
         context["types_display"] = self.get_ooi_types_display()
         context["object_type_filters"] = self.get_ooi_type_filters()
+        context["select_oois_form"] = SelectOOIForm(context["ooi_list"])
+        context["scan_levels"] = [alias for level, alias in SCAN_LEVEL.choices]
 
         return context
 
+    def post(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
+        """Perform bulk action on selected oois."""
+
+        # TODO:
+        #  - permissions
+        #  - Styled select field
+
+        selected_oois = request.POST.getlist("ooi")
+
+        if not selected_oois:
+            messages.add_message(request, messages.ERROR, _("No OOIs selected."))
+            return self.get(request, *args, **kwargs)
+
+        action = request.POST.get("action")
+
+        if action == "delete":
+            return self.delete_oois(selected_oois, request, *args, **kwargs)
+
+        if action == "update-scan-profile":
+            return self.set_scan_profiles(selected_oois, request, *args, **kwargs)
+
+        messages.add_message(request, messages.ERROR, _("Unknown action."))
+        return self.get(request, *args, **kwargs)
```

Comment:
(Not blocking/ interesting discussion for later) Although rendering the page using the same GET handler content including an error message is fine, I would still expect a 422 (or 400 even?) status code? @underdarknl

Reply:
Agreed, when posting with no sensible action an error should be shown. For the users convenience we could render the original form again, with the warning presented. But since your POST was invalid a 422 would be returned as the status code.

Reply:
Not sure what the Django-way to handle 422 responses from a form submission in a user-friendly way is

Reply:
Fixed